tool/formula/src/main/java/org/openscience/cdk/decomp/Alphabet.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  This file is part of the SIRIUS library for analyzing MS and MS/MS data
+ *
+ *  Copyright (C) 2013-2015 Kai Dührkop
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with SIRIUS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.openscience.cdk.decomp;
+
+/**
+ * The alphabet for which a given weight is decomposed. An alphabet is a vector c_1..c_k of k characters of Type T.
+ * It maps each character to a weight. It supports access by an index as well as by the character itself.
+ *
+ * @param <T> type of a single character in the alphabet
+ */
+interface Alphabet<T> {
```

Comment:
Is there likely to be different Alphabets? I'd be tempted to pull these methods up to the single ChemicalAlphabet implementation and drop the interface.